liquidprompt

```diff
@@ -1868,6 +1877,20 @@ _lp_set_prompt()
     fi
 }
 
+_remaining_space()
```

Comment:
This is pretty much implemented already in `_lp_as_text()`.

Besides, adding a `perl` dependency is a no-go. Not to mention I'm pretty sure this will only work for Bash, see how `_lp_as_text()` does this:

https://github.com/nojhan/liquidprompt/blob/deff598f30f097279d6f6959ba49441923dec041/liquidprompt#L1583-L1595